test/e2e/kubernetes/pod/pod.go

```diff
@@ -119,6 +123,35 @@ type Status struct {
 	ContainerStatuses []ContainerStatus `json:"containerStatuses"`
 }
 
+// ReplaceContainerImageFromFile loads in a YAML, finds the image: line, and replaces it with the value of containerImage
+func ReplaceContainerImageFromFile(filename, containerImage string) (string, error) {
+	var outString string
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Printf("Error opening source YAML file %s\n", filename)
+		return "", err
+	}
+	defer file.Close()
+	re := regexp.MustCompile("(image:) .*$")
+	replacementString := "$1 " + containerImage
+	reader := bufio.NewReader(file)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		outString += re.ReplaceAllString(scanner.Text(), replacementString) + "\n"
+	}
+	err = scanner.Err()
+	if err != nil {
+		return "", err
+	}
+	_, filenameOnly := path.Split(filename)
+	tmpFile, err := ioutil.TempFile(os.TempDir(), filenameOnly)
+	if err != nil {
+		return "", err
+	}
+	_, err = tmpFile.Write([]byte(outString))
+	return tmpFile.Name(), err
+}
+
 // CreatePodFromFile will create a Pod from file with a name
 func CreatePodFromFile(filename, name, namespace string, sleep, duration time.Duration) (*Pod, error) {
 	cmd := exec.Command("kubectl", "apply", "-f", filename)
```

Comment:
@tariq1890 `kubectl apply` can actually substitute as `kubectl create`. In this test it's used in place of create. I'd like to avoid creating a deployment, then patching it because if the wrong thing is deployed first - it will still be pulled which can be a 10GB mistake